qiskit/circuit/library/arithmetic/adder/ripple_carry_adder.py

```diff
@@ -0,0 +1,116 @@
+    Circuit to compute the sum of two qubit registers using the approach from [1].
+    Given two equally sized input registers that store quantum states
+    :math:`|a\rangle` and :math:`|b\rangle`, performs addition of numbers that
+    can be represented by the states, storing the resulting state in-place in the second register:
+
+    .. math::
+
+        |a\rangle |b\rangle \mapsto |a\rangle |a+b\rangle
+
+    Here :math:`|a\rangle` (and correspondingly :math:`|b\rangle`) stands for the direct product
+    :math:`|a_n\rangle \otimes |a_{n-1}\rangle \ldots |a_{1}\rangle \otimes |a_{0}\rangle`
+    which denotes a quantum register prepared with the value :math:`a = 2^{0}a_{0} + 2^{1}a_{1} +
+    \ldots 2^{n}a_{n}` [2].
+
+    As an example, a ripple-carry adder circuit that performs addition on two 3-qubit sized
+    registers is as follows:
+
+    .. parsed-literal::
+
+                     ┌──────┐┌──────┐                     ┌──────┐┌──────┐
+          input_a_0: ┤0     ├┤2     ├─────────────────────┤2     ├┤0     ├
+                     │      ││      │┌──────┐     ┌──────┐│      ││      │
+          input_a_1: ┤      ├┤0     ├┤2     ├─────┤2     ├┤0     ├┤      ├
+                     │      ││      ││      │     │      ││      ││      │
+          input_a_2: ┤      ├┤  MAJ ├┤0     ├──■──┤0     ├┤  UMA ├┤      ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_0: ┤1 MAJ ├┤      ├┤  MAJ ├──┼──┤  UMA ├┤      ├┤1 UMA ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_1: ┤      ├┤1     ├┤      ├──┼──┤      ├┤1     ├┤      ├
+                     │      │└──────┘│      │  │  │      │└──────┘│      │
+          input_b_2: ┤      ├────────┤1     ├──┼──┤1     ├────────┤      ├
+                     │      │        └──────┘  │  └──────┘        │      │
+         carry_in_0: ┤2     ├──────────────────┼──────────────────┤2     ├
+                     └──────┘                ┌─┴─┐                └──────┘
+        carry_out_0: ────────────────────────┤ X ├────────────────────────
+                                             └───┘
+
+    Here *MAJ* and *UMA* gates correspond to the gates introduced in [1]. Note that
+    in this implementation the input register qubits are ordered as all qubits from
+    the first input register, followed by all qubits from the second input register.
+    This is different ordering as compared to Figure 4 in [1], which leads to a different
+    drawing of the circuit.
+
+    **References:**
+
+    [1] Cuccaro et al., A new quantum ripple-carry addition circuit, 2004.
+    `arXiv:quant-ph/0410184 <https://arxiv.org/pdf/quant-ph/0410184.pdf>`_
+
+    [2] Vedral et al., Quantum Networks for Elementary Arithmetic Operations, 1995.
+    `arXiv:quant-ph/9511018 <https://arxiv.org/pdf/quant-ph/9511018.pdf>`_
+    """
+
+    def __init__(self,
+                 num_state_qubits: int,
+                 name: str = 'ripple_carry_adder'
```

Comment:
Do you mind renaming it like this? I think this looks nicer 😄 
```suggestion
                 name: str = 'RippleCarryAdder'
```

Reply:
Thanks, I will keep in mind that it's CamelCase for circuit names :D Is it fine to keep underscore_case for registers?

Reply:
yes 👍🏻

Reply:
I think everything is clear here now, so I will mark the conversation as resolved for the time being :)